src/DataContainer/PaletteManipulator.php

```diff
@@ -87,6 +92,22 @@ public function addField($name, $parent, string $position = self::POSITION_AFTER
         return $this;
     }
 
+    /**
+     * Parent refers to the legend. If $parent is empty the field is removed everywhere.
+     *
+     * @param string|array               $name
+     * @param string|array               $parent
+     */
+    public function removeField($name, $parent = []): self
```

Comment:
To be consistent, `$parent` should not have a default value

Reply:
That doesn't really make sense to me. When adding a field, you always want to have a parent. Maybe a legend or another field depending on the position value.

The logic behind removing a field is different. Furhtermore as a parent there is only a legend accepted. Removing a field without knowing the legend makes sense to me. But I'd happy with renaming the param to `$legend` instead to make it more explicit.

Reply:
You're right. Shouldn't the default value then be `null`?

Reply:
As it's converted to an array later on I already used an empty array as default value.

https://github.com/contao/core-bundle/blob/c120c7a76069927a80a1afc1061af7c9e1763cc4/src/DataContainer/PaletteManipulator.php#L105

But I don't mind to change it. It's imho a matter of taste anyway.

Reply:
I agree, I don't really like any of the options 😄 
Thinking of it, do we really need an option to actually set a palette? I mean would you ever want to a remove field if it's in palette A but not if it's in palette B? That sounds very strange to me. So maybe we can just drop that argument?

Reply:
As I explained in the opening post, it's required if I want to move a field from one legend to another (Yeah, it's a real project use case) without creating two instances of the palette manipulator, f.e.:

```php
PaletteManipulator::create()
    ->removeField('published', 'publish_legend')
    ->addField('published', 'title_legend', PaletteManipulator::POSITION_APPEND);
```

Reply:
I see. Does it ever make sense to have the same field in multiple legends? Like if you add it twice, should the palette manipulator ignore that or should the later win?

Reply:
No, it doesn't make sense to add a field twice and probably causes afair some issues.It could happen when multiple extensions change the dca. 

Imho there's no need to handle such cases. ATM the palette manipulator has a low level implementation as it does not set any constraints. F.e. you could argue that adding a field which does not exist in the dca shouldn't be allowed as well. Such constraints does not exist so far and I think it's not the purpose of the manipulator.

Reply:
Isn't that the case with tl_content singleSrc for example?
If the type is "Hyperlink" the legend is "Image link settings" - if the type is "Image" the legend is"Files and folders".
You might want to remove one but keep the other or am i mixing up paletes and legends now 😕 ?.

Reply:
That would be separate palettes then, which is a different case and totally supported.